--- a/python_dsl_template.jinja2	2024-05-22 11:58:26.259019305 -0400
+++ b/python_dsl_template.jinja2	2024-05-22 12:00:31.507205520 -0400
@@ -18,7 +18,6 @@
 from kfp.kubernetes import common
 from kfp.kubernetes import kubernetes_executor_config_pb2 as pb
 
-from kfp.kubernetes import add_toleration, add_pod_label, add_pod_annotation
 
 {# Load statements for custom components                            -#}
 {# component_hash = """<TEXT>"""                                  -#}
@@ -64,9 +63,6 @@
     secret.use_secret_as_env({{ task_name }}, "{{ workflow_task.task_modifiers.object_storage_secret }}", { "AWS_ACCESS_KEY_ID": "AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY": "AWS_SECRET_ACCESS_KEY" })
 {%  endif %}
     {{ task_name }}.set_display_name("{{ workflow_task.name | string_delimiter_safe }}")
-{%  if workflow_task.doc %}
-    add_pod_annotation({{ task_name }}, "elyra/node-user-doc","""{{ workflow_task.doc| string_delimiter_safe }}""")
-{%  endif %}
 {%  if workflow_task.task_modifiers.cpu_request %}
     {{ task_name }}.set_cpu_request(cpu="{{ workflow_task.task_modifiers.cpu_request }}")
 {%  endif %}
@@ -93,16 +89,6 @@
 {%  if workflow_task.task_modifiers.disable_node_caching %}
     {{ task_name }}.execution_options.caching_strategy.max_cache_staleness = "P0D"
 {%  endif %}
-{%  if workflow_task.task_modifiers.pod_labels %}
-{%    for pod_label_key, pod_label_value in workflow_task.task_modifiers.pod_labels.items() %}
-    add_pod_label({{ task_name }}, "{{ pod_label_key }}", "{{ pod_label_value }}")
-{%    endfor %}
-{%  endif %}
-{%  if workflow_task.task_modifiers.pod_annotations %}
-{%    for pod_annotation_key, pod_annotation_value in workflow_task.task_modifiers.pod_annotations.items() %}
-    add_pod_annotation({{ task_name }}, "{{ pod_annotation_key }}" , """{{ pod_annotation_value | string_delimiter_safe }}""")
-{%    endfor %}
-{%  endif %}
 {%  if workflow_task.task_modifiers.kubernetes_secrets %}
 {%    for env_var, secret_dict in workflow_task.task_modifiers.kubernetes_secrets.items() %}
     secret.use_secret_as_env({{ task_name }}, "{{ secret_dict.name }}", { "{{ secret_dict.key }}" : "{{ env_var }}" })
@@ -113,29 +99,6 @@
     volume.mount_pvc({{ task_name }}, "{{ volume_dict.pvc_name }}", "{{ volume_path }}")
 {%    endfor %}
 {%  endif %}
-{%  if workflow_task.task_modifiers.kubernetes_tolerations %}
-{% for toleration_dict in workflow_task.task_modifiers.kubernetes_tolerations.values() %}
-    add_toleration(
-        {{ task_name }},
-        {% if toleration_dict.key %}
-            key="{{ toleration_dict.key }}",
-        {% else %}
-            key=None,
-        {% endif %}
-        operator="{{ toleration_dict.operator }}",
-        {% if toleration_dict.value %}
-            value="{{ toleration_dict.value | string_delimiter_safe }}",
-        {% else %}
-            value=None,
-        {% endif %}
-        {% if toleration_dict.effect %}
-            effect="{{ toleration_dict.effect }}",
-        {% else %}
-            effect=None,
-        {% endif %}
-    )
-{% endfor %}
-{%  endif %}
 {#  declare upstream dependencies -#}
 {%  if workflow_task.upstream_workflow_task_ids %}
 {%    for upstream_workflow_task_id in workflow_task.upstream_workflow_task_ids %}
