diff --git a/ci/build/build-standalone-release.sh b/ci/build/build-standalone-release.sh
index b0833db81..6f880b7cc 100755
--- a/ci/build/build-standalone-release.sh
+++ b/ci/build/build-standalone-release.sh
@@ -13,6 +13,14 @@ main() {
   rsync "$RELEASE_PATH/" "$RELEASE_PATH-standalone"
   RELEASE_PATH+=-standalone
 
+  VSCODE_SRC_PATH="lib/vscode"
+  jq --slurp '. as $in | ($in[0] + $in[1]) | .dependencies = (($in[0].dependencies // {}) + ($in[1].dependencies // {}))' \
+    "$VSCODE_SRC_PATH/remote/package.json" \
+    "$RELEASE_PATH/lib/vscode/package.json" > "$RELEASE_PATH/lib/vscode/package.json.merged"
+  mv "$RELEASE_PATH/lib/vscode/package.json.merged" "$RELEASE_PATH/lib/vscode/package.json"
+  cp "$VSCODE_SRC_PATH/remote/package-lock.json" "$RELEASE_PATH/lib/vscode/npm-shrinkwrap.json"
+  cp ./ci/build/npm-postinstall.sh "$RELEASE_PATH/postinstall.sh"
+
   # Package managers may shim their own "node" wrapper into the PATH, so run
   # node and ask it for its true path.
   local node_path
@@ -25,6 +33,13 @@ main() {
 
   chmod 755 "$RELEASE_PATH/lib/node"
 
+  if [ -f /root/scripts/lockfile-generators/rewrite-cachi2-path.sh ]; then
+    . /root/scripts/lockfile-generators/rewrite-cachi2-path.sh
+    rewrite_cachi2_path "$RELEASE_PATH/npm-shrinkwrap.json"
+    rewrite_cachi2_path "$RELEASE_PATH/lib/vscode/npm-shrinkwrap.json"
+    rewrite_cachi2_path "$RELEASE_PATH/lib/vscode/extensions/npm-shrinkwrap.json"
+  fi
+
   pushd "$RELEASE_PATH"
   npm install --unsafe-perm --omit=dev
   # Code deletes some files from the extension node_modules directory which
