--- a/processor_kfp.py	2023-06-09 10:19:08.882563609 -0400
+++ b/processor_kfp.py	2023-07-13 19:31:43.572407879 -0400
@@ -213,6 +213,7 @@
                     credentials=auth_info.get("credentials", None),
                     existing_token=auth_info.get("existing_token", None),
                     namespace=user_namespace,
+                    ssl_ca_cert=auth_info.get("ssl_ca_cert", None),
                 )
             else:
                 client = ArgoClient(
@@ -309,8 +310,16 @@
 
                 # Create an instance id that will be used to store
                 # the pipelines' dependencies, if applicable
-                pipeline_instance_id = f"{pipeline_name}-{timestamp}"
+                pipeline_instance_id = ""
 
+                # Use environment variable to set version as suffix instead of timestamp
+                if os.environ.get("KFP_SUFFIX_USE_VERSION") == "true" :
+                    # Version is determined by the count of existing pipeline versions
+                    version = client.list_pipeline_versions(pipeline_id=pipeline_id).total_size
+                    pipeline_instance_id = f"{pipeline_name}-v{version}"
+                else :
+                    pipeline_instance_id = f"{pipeline_name}-{timestamp}"
+
                 # Generate Python DSL from workflow
                 pipeline_dsl = self._generate_pipeline_dsl(
                     pipeline=pipeline,
@@ -416,7 +425,7 @@
 
                 # create pipeline run (or specified pipeline version)
                 run = client.run_pipeline(
-                    experiment_id=experiment.id, job_name=job_name, pipeline_id=pipeline_id, version_id=version_id
+                    experiment_id=experiment.id, job_name=job_name, pipeline_id=pipeline_id, version_id=pipeline_id
                 )
 
             except Exception as ex:
@@ -435,7 +444,7 @@
 
             self.log_pipeline_info(
                 pipeline_name,
-                f"pipeline submitted: {public_api_endpoint}/#/runs/details/{run.id}",
+                f"pipeline submitted: {public_api_endpoint}/{run.id}",
                 duration=time.time() - t0,
             )
 
@@ -451,7 +460,7 @@
 
         return KfpPipelineProcessorResponse(
             run_id=run.id,
-            run_url=f"{public_api_endpoint}/#/runs/details/{run.id}",
+            run_url=f"{public_api_endpoint}/{run.id}",
             object_storage_url=object_storage_url,
             object_storage_path=object_storage_path,
         )